### PR TITLE
LPS-123777 Create tests related to hide/show repetable button

### PR DESCRIPTION
**Pull request by alinedoleron**
Related Issue: https://issues.liferay.com/browse/LPS-123777

### CODE REVIEW

**Comment by liferay-forms** [comment]
ci:test:sf

**Comment by liferay-forms** [comment]
Hi, you don’t need to run the CI commands anymore, I’ll do it for you and let you know if any issue happens.

**Comment by liferay-forms** [comment]
ci:test:relevant

**Comment by liferay-forms** [comment]
Hey @alinedoleron, can you please check if these failures are related to your changes? Thanks

**Comment by alinedoleron** [comment]
ci:test:relevant

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/DocumentLibrary/DocumentLibrary.es.js [hunk 1]
```diff
@@ -300,13 +301,27 @@ const Main = ({
...
+
+		let repetitionsCounter = 0;
+
+		visitor.mapFields((field) => {
+			if (fieldName === field.fieldName) {
+				repetitionsCounter++;
+			}
+		});
+
+		return repetitionsCounter == maximumRepetitions;
```
Comment (natocesarrego) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/DocumentLibrary/DocumentLibrary.es.js:322]: @alinedoleron, let's use `===` instead, ok?

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/test/js/FieldBase/ReactFieldBase.es.js [hunk 2]
```diff
@@ -146,4 +146,37 @@ describe('ReactFieldBase', () => {
 
 		expect(container).toMatchSnapshot();
 	});
+
+	it('renders the the add when repeatable is true', () => {
```
Comment (natocesarrego) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/test/js/FieldBase/ReactFieldBase.es.js:150]: @alinedoleron, the word `the` is repeated. Maybe we should say `renders the add button when repeatable is true`.

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/test/js/FieldBase/ReactFieldBase.es.js [hunk 3]
```diff
@@ -146,4 +146,37 @@ describe('ReactFieldBase', () => {
...
+			/>
+		);
+		act(() => {
+			jest.runAllTimers();
+		});
+
+		expect(container).toMatchSnapshot();
+	});
+
+	it('does not render the the add when repeatable is true and the maximum limit of repetions is reached', () => {
```
Comment (natocesarrego) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/test/js/FieldBase/ReactFieldBase.es.js:166]: @alinedoleron, same as previous comment.

**Review by natocesarrego — CHANGES_REQUESTED** [review]
Good job, @carolmariaabb and @alinedoleron! 😄

**Pull request closed without merging**